### PR TITLE
task 2

### PR DESCRIPTION
**Pull request by Odinachi**

### CODE REVIEW

File: lib/main.dart [hunk 1]
```diff
@@ -4,29 +4,52 @@ import 'package:flutter/material.dart';
...
     options: DefaultFirebaseOptions.currentPlatform,
   );
-  runApp(const MyApp());
+  runApp(MyApp());
 }
 
 class MyApp extends StatelessWidget {
-  const MyApp({Key? key}) : super(key: key);
-  static User? user = FirebaseAuth.instance.currentUser;
+  MyApp({Key? key}) : super(key: key);
```
Comment (lesnitsky) [inline lib/main.dart:19]: why did you make this widget not const?

File: lib/screens/auth_screen.dart [hunk 2]
```diff
@@ -1,32 +1,30 @@
...
+      resizeToAvoidBottomInset: true,
+      providerConfigs: const [
+        EmailProviderConfiguration(),
+      ],
+      actions: [
+        AuthStateChangeAction((context, AuthState state) {
+          if (state is SignedIn) {
+            //after user is logged in we want to determine whether their email is verified or not and then move them accordingly
+            if (state.user?.emailVerified == true) {
+              context.go("/home");
```
Comment (lesnitsky) [inline lib/screens/auth_screen.dart:21]: is it possible to go back after `go`? I'm not really familiar with go_router, but I'm assuming there should be a way to navigate to a new route without being able to go back (if that's the case)

File: lib/main.dart (hunk 1 above)
Comment (Odinachi) [reply lib/main.dart:19]: I think the compiler asked me to.

File: lib/main.dart [hunk 3]
```diff
@@ -4,29 +4,52 @@ import 'package:flutter/material.dart';
 import 'package:flutterfire_test/firebase_options.dart';
 import 'package:flutterfire_test/screens/auth_screen.dart';
 import 'package:flutterfire_test/screens/home_screen.dart';
+import 'package:flutterfire_test/screens/verify_email_screen.dart';
```
[the remark this answers is not in the document]
Comment (Odinachi) [reply lib/main.dart:7]: it's a screen I created, so the screen houses the verify email widget

File: lib/screens/auth_screen.dart [hunk 4]
```diff
@@ -1,32 +1,30 @@
...
-                  context, "home", (route) => false);
-            });
+    return SignInScreen(
+      oauthButtonVariant: OAuthButtonVariant.icon,
+      resizeToAvoidBottomInset: true,
+      providerConfigs: const [
+        EmailProviderConfiguration(),
+      ],
+      actions: [
+        AuthStateChangeAction((context, AuthState state) {
```
[the remark this answers is not in the document]
Comment (Odinachi) [reply lib/screens/auth_screen.dart:17]: it does but I felt it was a good practice.

File: lib/screens/auth_screen.dart (hunk 2 above)
Comment (Odinachi) [reply lib/screens/auth_screen.dart:21]: alright, Let me check.

File: lib/main.dart (hunk 3 above)
[the remark this answers is not in the document]
Comment (lesnitsky) [reply lib/main.dart:7]: oh my bad, I'm used to matching folder and package name :)

File: lib/main.dart (hunk 1 above)
Comment (lesnitsky) [reply lib/main.dart:19]: image

File: lib/screens/auth_screen.dart (hunk 4 above)
[the remark this answers is not in the document]
Comment (lesnitsky) [reply lib/screens/auth_screen.dart:17]: I think its preferred to omit types if compiler is able to infer it.

[the remark this answers is not in the document]
Comment (Odinachi) [reply lib/screens/auth_screen.dart:17]: Alright.

File: lib/screens/auth_screen.dart [hunk 5]
```diff
@@ -1,32 +1,30 @@
...
+          if (state is SignedIn) {
+            //after user is logged in we want to determine whether their email is verified or not and then move them accordingly
+            if (state.user?.emailVerified == true) {
```
[the remark this answers is not in the document]
Comment (Odinachi) [reply lib/screens/auth_screen.dart:20]: this I noted.

**Pull request merged**